### PR TITLE
M5198 One click importer

### PR DESCRIPTION
**Pull request by Mark-de-Haan**
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] (If you have changed REST API interface) view-swagger.ftl updated
- [ ] Test plan template updated
- [ ] Clean commits

### CODE REVIEW

File: molgenis-one-click-importer/src/main/frontend/package.json [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  },
+  "dependencies": {
+    "@molgenis/molgenis-i18n-js": "^0.0.5",
+    "bootstrap": "4.0.0-alpha.6",
+    "bootstrap-vue": "^0.18.0",
+    "font-awesome": "^4.7.0",
+    "isomorphic-fetch": "^2.2.1",
+    "vue": "^2.2.6",
+    "vue-router": "^2.3.1",
+    "vue-sweetalert": "^0.1.17",
```
Comment (ConnorStroomberg) [inline molgenis-one-click-importer/src/main/frontend/package.json]: we should remove the unused deps

File: molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/controller/OneClickImportResponse.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package org.molgenis.oneclickimporter.controller;
+
+import com.google.auto.value.AutoValue;
+import org.molgenis.gson.AutoGson;
+
+@AutoValue
+@AutoGson(autoValueClass = AutoValue_OneClickImportResponse.class)
+public abstract class OneClickImportResponse
```
Comment (ConnorStroomberg) [inline molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/controller/OneClickImportResponse.java]: i dont think this class is used anymore

File: molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/service/Impl/AttributeTypeServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+		while (currentRowIndex < rowCount && !guessCompleted)
+		{
+			Object value = dataValues.get(currentRowIndex);
+			AttributeType basicType = getBasicAttributeType(value);
+
+			AttributeType basicTypeGuess = getCommonType(currentGuess, basicType);
+			AttributeType enrichedTypeGuess = getEnrichedType(basicTypeGuess, value);
+
+			// If the newly found type is not narrower the current type do not update as the value wil not fit
+			// ( for example; as string does not fit into a integer).
```
Comment (ConnorStroomberg) [inline molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/service/Impl/AttributeTypeServiceImpl.java]: type as should be a

File: molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/service/Impl/EntityServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+		// Check if first column can be used as id ( has unique values )
+		List<Column> columns = dataCollection.getColumns();
+		Column firstColumn = columns.get(0);
+		final boolean isFirstColumnUnique = oneClickImporterService.hasUniqueValues(firstColumn);
+
+		AttributeType type = attributeTypeService.guessAttributeType(firstColumn.getDataValues());
+		final boolean isValidAttributeType = AttributeUtils.getValidIdAttributeTypes().contains(type);
+		final boolean useAutoId = !isFirstColumnUnique || !isValidAttributeType;
+
+		if (useAutoId)
```
Comment (ConnorStroomberg) [inline molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/service/Impl/EntityServiceImpl.java]: rewrite use '?' and split out id var

File: molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/service/Impl/EntityServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+			}
+		});
+
+		// Store the dataTable (metadata only)
+		permissionSystemService.giveUserWriteMetaPermissions(entityType);
+		metaDataService.addEntityType(entityType);
+
+		AtomicInteger rowIndex = new AtomicInteger(0);
+		List<Entity> rows = newArrayList();
+		while (rowIndex.get() < dataCollection.getColumns().get(0).getDataValues().size())
```
Comment (ConnorStroomberg) [inline molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/service/Impl/EntityServiceImpl.java]: use for

File: molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/service/Impl/ExcelServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+{
+	@Override
+	public List<Sheet> buildExcelSheetsFromFile(File file)
+			throws IOException, InvalidFormatException, NoDataException, EmptySheetException
+	{
+		Workbook workbook = WorkbookFactory.create(file);
+		int numberOfSheets = workbook.getNumberOfSheets();
+
+		List<Sheet> sheets = newArrayList();
+		for (int index = 0; index < numberOfSheets; index++)
```
Comment (ConnorStroomberg) [inline molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/service/Impl/ExcelServiceImpl.java]: maybe use iterator

Comment (Mark-de-Haan) [inline molgenis-one-click-importer/src/main/java/org/molgenis/oneclickimporter/service/Impl/ExcelServiceImpl.java]: Can't

> A Workbook doesn't implement Iterable<Sheet> like a Sheet implements Iterable<Row>, so you can't use an "enhanced" for loop. But you can loop through the Sheet objects yourself, with a traditional for loop. Use getNumberOfSheets() to find the number of sheets, and getSheetAt to retrieve the individual Sheet objects.

@https://stackoverflow.com/questions/20340915/how-to-go-though-the-sheets-of-a-workbook-with-apache-poi

**Comment by ConnorStroomberg** [comment]
test this please

**Pull request merged**